### PR TITLE
feat(python): Support use of SQLAlchemy "Connectable" in `write_database`

### PR DESCRIPTION
**Pull request by phi-friday**
`pandas` supports `sa.engine.Connectable`.
Passing `sa.engine.Connectable` allows the user to control the transaction.

### CODE REVIEW

**Comment by alexander-beedie** [comment]
I will double-check tonight (on mobile at the moment), but from what I can see this looks like a good addition; thanks!

File: py-polars/tests/unit/io/database/test_write.py [hunk 1]
```diff
@@ -259,3 +261,60 @@ def test_write_database_using_sa_session(tmp_path: str) -> None:
         )
 
     assert_frame_equal(result, df)
+
+
+@pytest.mark.write_disk()
+@pytest.mark.parametrize("pass_connection", [True, False])
```
Comment (alexander-beedie) [inline py-polars/tests/unit/io/database/test_write.py:267]:
```suggestion
@pytest.mark.parametrize("pass_connection", [True, False])
@pytest.mark.skipif(
    parse_version(sqlalchemy.__version__) < (2, 0),
    reason="requires sqlalchemy >= 2.0",
)
```

File: py-polars/tests/unit/io/database/test_write.py [hunk 2]
```diff
@@ -259,3 +261,60 @@ def test_write_database_using_sa_session(tmp_path: str) -> None:
...
+        count = pl.read_database(
+            query=f"select count(*) from {table_name}", connection=session
+        ).item(0, 0)
+
+    assert isinstance(count, int)
+    assert count == 0
+
+
+@pytest.mark.write_disk()
+@pytest.mark.parametrize("pass_connection", [True, False])
```
Comment (alexander-beedie) [inline py-polars/tests/unit/io/database/test_write.py:296]:
```suggestion
@pytest.mark.parametrize("pass_connection", [True, False])
@pytest.mark.skipif(
    parse_version(sqlalchemy.__version__) < (2, 0),
    reason="requires sqlalchemy >= 2.0",
)
```

**Review by alexander-beedie — CHANGES_REQUESTED** [review]
Let's version-gate the tests, as they don't work on earlier versions of SQLAlchemy, which I sometimes check, as not everyone has moved to v2.0 yet. (The two suggested changes require an additional import `from polars._utils.various import parse_version` at the top of the test module).

Otherwise, looks good to me 👍

File: py-polars/tests/unit/io/database/test_write.py (hunk 1 above)
Comment (phi-friday) [reply py-polars/tests/unit/io/database/test_write.py:265]: Fixed test to work with `sqlalchemy<2`.

File: py-polars/tests/unit/io/database/test_write.py (hunk 2 above)
Comment (phi-friday) [reply py-polars/tests/unit/io/database/test_write.py:295]: Fixed test to work with `sqlalchemy<2`.

**Comment by phi-friday** [comment]
Since it's entirely up to the user to control the transaction, I decided it was okay to just modify the test.

**Comment by phi-friday** [comment]
This is unrelated to this pr, but when I lowered the versions of `sqlalchemy<2` and `pandas<2.2`, I got 3 errors.

```log
FAILED tests/unit/interchange/test_roundtrip.py::test_from_dataframe_pandas_timestamp_ns - AssertionError: DataFrames are different (value mismatch for column 'a')
FAILED tests/unit/io/database/test_read.py::test_read_database_parameterised - sqlalchemy.exc.RemovedIn20Warning: Deprecated API features detected! These feature(s) are not co...
ERROR tests/unit/io/database/test_async.py
```

**Comment by alexander-beedie** [comment]
> Since it's entirely up to the user to control the transaction, I decided it was okay to just modify the test.

Equally valid ;)

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**